src/shared-components/chip/index.tsx

```diff
@@ -1,27 +1,95 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { useTheme } from 'emotion-theming';
 
 import { ChipStyles, ChipText } from './style';
+import { ThemeColors, ThemeType } from '../../constants';
 
-export type StatusType = 'default' | 'success' | 'error' | 'secondary';
+export type StatusType = 'primary' | 'success' | 'error' | 'white';
 
 export interface ChipProps {
+  isLowContrast?: boolean;
   status?: StatusType;
   text: string;
 }
 
+interface GetStylesProps extends Required<Omit<ChipProps, 'text'>> {
+  theme: ThemeType;
+}
+
+const getStyles = ({ isLowContrast, status, theme }: GetStylesProps) => {
```

Comment:
A lot of our more recent `theme`-related changes have lifted up color/etc. logic to the component definition instead of nesting it away in the `style` file. I like this paradigm, and I think it makes it easier for us to identify places where we can optimize our `emotion` usage by avoiding nesting functions and objects within our `style` definitions. 

Despite the net positive lines of code added between `index.tsx` and `style.ts`, we actually overall [decreased the size of our bundle, according to bundlewatch.](https://service.bundlewatch.io/results?d=details%7CrepoOwner%7Ccurology%7CrepoName%7Cradiance-ui%7CrepoCurrentBranch%7Cchip/low-contrast%7CrepoBranchBase%7Cmaster%7CcommitSha%7C91558353b4c6c96b78ff15edb9f6f96e656860bb%7Cresults%7Cstatus%7Cpass%7CfullResults%7CfilePath%7Cdist/bundle.js%7Csize%7CbaseBranchSize%7CmaxSize%7Cdist/bundle-es/shared-components/accordion/index.js%7Cdist/bundle-es/shared-components/accordion/thumbnails/index.js%7Cdist/bundle-es/shared-components/alert/index.js%7Cdist/bundle-es/shared-components/avatar/index.js%7Cdist/bundle-es/shared-components/banner/index.js%7Cdist/bundle-es/shared-components/button/components/anchorLinkButton/index.js%7Cdist/bundle-es/shared-components/button/components/linkButton/index.js%7Cdist/bundle-es/shared-components/button/components/roundButton/index.js%7Cdist/bundle-es/shared-components/button/components/textButton/index.js%7Cdist/bundle-es/shared-components/button/index.js%7Cdist/bundle-es/shared-components/button/shared-components/container/index.js%7Cdist/bundle-es/shared-components/button/shared-components/loader/index.js%7Cdist/bundle-es/shared-components/callout/index.js%7Cdist/bundle-es/shared-components/carousel/arrow/index.js%7Cdist/bundle-es/shared-components/carousel/index.js%7Cdist/bundle-es/shared-components/checkbox/index.js%7Cdist/bundle-es/shared-components/chip/index.js%7Cdist/bundle-es/shared-components/dialogModal/index.js%7Cdist/bundle-es/shared-components/dropdown/index.js%7Cdist/bundle-es/shared-components/field/index.js%7Cdist/bundle-es/shared-components/icon/index.js%7Cdist/bundle-es/shared-components/immersiveModal/index.js%7Cdist/bundle-es/shared-components/indicator/index.js%7Cdist/bundle-es/shared-components/loadingSpinner/index.js%7Cdist/bundle-es/shared-components/offClickWrapper/index.js%7Cdist/bundle-es/shared-components/optionButton/index.js%7Cdist/bundle-es/shared-components/progressBar/index.js%7Cdist/bundle-es/shared-components/radioButton/index.js%7Cdist/bundle-es/shared-components/selectorButton/index.js%7Cdist/bundle-es/shared-components/tabs/index.js%7Cdist/bundle-es/shared-components/toggle/index.js%7Cdist/bundle-es/shared-components/tooltip/index.js%7Cdist/bundle-es/shared-components/transitions/index.js%7Cdist/bundle-es/shared-components/verificationMessages/formatMessage/index.js%7Cdist/bundle-es/shared-components/verificationMessages/index.js%7Cdist/bundle-es/shared-components/typography/index.js%7Csummary%7CEverything%2Bis%2Bin%2Bcheck%2B(%252B1.61KB,%2B-1.71KB,%2B-0.0%2525)%5EGAX3%7CGC9Q%7CLY4G%7C2IM%7C2IM%7CFSW%7CW3%7CW3%7CFSW%7C4YE%7C4YE%7CFSW%7CFB%7CFB%7CFSW%7C13Y%7C13Y%7CFSW%7COX%7COX%7CFSW%7C28F%7C28F%7CFSW%7C2X7%7C2X7%7CFSW%7C154%7C154%7CFSW%7C2XF%7C2XF%7CFSW%7C2LK%7C2LK%7CFSW%7CNT%7CNT%7CFSW%7C1I8%7C1I8%7CFSW%7C17L%7C17L%7CFSW%7C460%7C460%7CFSW%7C1OM%7C1OM%7CFSW%7C1XS%7CO4%7CFSW%7C3F8%7C3F8%7CFSW%7C35T%7C35T%7CFSW%7C254%7C254%7CFSW%7C1C2%7C1C2%7CFSW%7C5NI%7C5NI%7CFSW%7C1CP%7C1CP%7CFSW%7C1BA%7C1BA%7CFSW%7CYT%7CYT%7CFSW%7C1MM%7C1MM%7CFSW%7C1MG%7C1MG%7CFSW%7C1JF%7C1JF%7CFSW%7C7Y4%7C7Y4%7CFSW%7C11R%7C11R%7CFSW%7C1E5%7C1E5%7CFSW%7C3YZ%7C3YZ%7CFSW%7CKB%7CKB%7CFSW%7CEO%7CEO%7CFSW%7C1E3%7C1E3%7CFSW%7C1595%7C1595%7C270G%5E%5E$0%7C$1%7C2%7C3%7C4%7C5%7C6%7C7%7C8%7C9%7CA%5D%7CB%7C$C%7CD%7CE%7C@$F%7CG%7CC%7CD%7CH%7C1M%7CI%7C1N%7CJ%7C1O%5D%7C$F%7CK%7CC%7CD%7CH%7C1P%7CI%7C1Q%7CJ%7C1R%5D%7C$F%7CL%7CC%7CD%7CH%7C1S%7CI%7C1T%7CJ%7C1U%5D%7C$F%7CM%7CC%7CD%7CH%7C1V%7CI%7C1W%7CJ%7C1X%5D%7C$F%7CN%7CC%7CD%7CH%7C1Y%7CI%7C1Z%7CJ%7C20%5D%7C$F%7CO%7CC%7CD%7CH%7C21%7CI%7C22%7CJ%7C23%5D%7C$F%7CP%7CC%7CD%7CH%7C24%7CI%7C25%7CJ%7C26%5D%7C$F%7CQ%7CC%7CD%7CH%7C27%7CI%7C28%7CJ%7C29%5D%7C$F%7CR%7CC%7CD%7CH%7C2A%7CI%7C2B%7CJ%7C2C%5D%7C$F%7CS%7CC%7CD%7CH%7C2D%7CI%7C2E%7CJ%7C2F%5D%7C$F%7CT%7CC%7CD%7CH%7C2G%7CI%7C2H%7CJ%7C2I%5D%7C$F%7CU%7CC%7CD%7CH%7C2J%7CI%7C2K%7CJ%7C2L%5D%7C$F%7CV%7CC%7CD%7CH%7C2M%7CI%7C2N%7CJ%7C2O%5D%7C$F%7CW%7CC%7CD%7CH%7C2P%7CI%7C2Q%7CJ%7C2R%5D%7C$F%7CX%7CC%7CD%7CH%7C2S%7CI%7C2T%7CJ%7C2U%5D%7C$F%7CY%7CC%7CD%7CH%7C2V%7CI%7C2W%7CJ%7C2X%5D%7C$F%7CZ%7CC%7CD%7CH%7C2Y%7CI%7C2Z%7CJ%7C30%5D%7C$F%7C10%7CC%7CD%7CH%7C31%7CI%7C32%7CJ%7C33%5D%7C$F%7C11%7CC%7CD%7CH%7C34%7CI%7C35%7CJ%7C36%5D%7C$F%7C12%7CC%7CD%7CH%7C37%7CI%7C38%7CJ%7C39%5D%7C$F%7C13%7CC%7CD%7CH%7C3A%7CI%7C3B%7CJ%7C3C%5D%7C$F%7C14%7CC%7CD%7CH%7C3D%7CI%7C3E%7CJ%7C3F%5D%7C$F%7C15%7CC%7CD%7CH%7C3G%7CI%7C3H%7CJ%7C3I%5D%7C$F%7C16%7CC%7CD%7CH%7C3J%7CI%7C3K%7CJ%7C3L%5D%7C$F%7C17%7CC%7CD%7CH%7C3M%7CI%7C3N%7CJ%7C3O%5D%7C$F%7C18%7CC%7CD%7CH%7C3P%7CI%7C3Q%7CJ%7C3R%5D%7C$F%7C19%7CC%7CD%7CH%7C3S%7CI%7C3T%7CJ%7C3U%5D%7C$F%7C1A%7CC%7CD%7CH%7C3V%7CI%7C3W%7CJ%7C3X%5D%7C$F%7C1B%7CC%7CD%7CH%7C3Y%7CI%7C3Z%7CJ%7C40%5D%7C$F%7C1C%7CC%7CD%7CH%7C41%7CI%7C42%7CJ%7C43%5D%7C$F%7C1D%7CC%7CD%7CH%7C44%7CI%7C45%7CJ%7C46%5D%7C$F%7C1E%7CC%7CD%7CH%7C47%7CI%7C48%7CJ%7C49%5D%7C$F%7C1F%7CC%7CD%7CH%7C4A%7CI%7C4B%7CJ%7C4C%5D%7C$F%7C1G%7CC%7CD%7CH%7C4D%7CI%7C4E%7CJ%7C4F%5D%7C$F%7C1H%7CC%7CD%7CH%7C4G%7CI%7C4H%7CJ%7C4I%5D%7C$F%7C1I%7CC%7CD%7CH%7C4J%7CI%7C4K%7CJ%7C4L%5D%7C$F%7C1J%7CC%7CD%7CH%7C4M%7CI%7C4N%7CJ%7C4O%5D%5D%7C1K%7C1L%5D%5D)